File: Backend/src/main/java/com/example/backend/entity/ArticleEntity.java
package com.example.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.type.SqlTypes;

import java.time.LocalDateTime;

@Entity
@Table(name = "articles")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class ArticleEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(name = "category_id", nullable = false)
    private Integer categoryId;

    @JdbcTypeCode(SqlTypes.JSON)
    @Column(name = "main_words", columnDefinition = "JSONB")
    private String mainWords;

    @Column(length = 50)
    private String author;

    @Column(name = "image_url", length = 250)
    private String imageUrl;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "likes_count", nullable = false)
    private Integer likesCount = 0;

    @PrePersist
    protected void onCreate() {
        LocalDateTime now = LocalDateTime.now();
        if (createdAt == null) createdAt = now;
        if (updatedAt == null) updatedAt = now;
        if (likesCount == null) likesCount = 0;
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}

File: MyApplication/data/src/main/java/com/example/data/Hints/model/ArticleEntity.kt
package com.example.data.Hints.model

import com.example.data.Hints.model.CategoryEntity
import com.example.domain.Hints.model.Article
import java.time.LocalDateTime

data class ArticleEntity(
    val id: Int,
    val title: String,
    val categoryId: Int,
    val mainWords: List<String>,
    val author: String,
    val imageUrl: String?,
    val createdAt: LocalDateTime,
    val updatedAt: LocalDateTime
) {
    fun toDomain(category: CategoryEntity, likesCount: Int): Article {
        return Article(
            id = id,
            title = title,
            category = category.toDomain(),
            mainWords = mainWords,
            author = author,
            imageUrl = imageUrl,
            createdAt = createdAt,
            updatedAt = updatedAt,
            likesCount = likesCount
        )
    }
}